### PR TITLE
Fixing a few things with signups

### PR DESCRIPTION
**Pull request by deadlybutter**
This should _hopefully_ be the end of all the bugginess. 

- Dispatched the `completed event` before the action itself, this is because if we have redirects it will prevent the completed event from firing and thus turning into an infinite loop.
- Properly created a new redux state instead of modifying it
- Set more variables in the signups reducer that was missing before
- Changed the event logic to be more clear

### CODE REVIEW

File: resources/assets/actions/event.js [hunk 1]
```diff
@@ -24,23 +24,26 @@ export function startQueue() {
     const queue = getArray(getDeviceId(), EVENT_STORAGE_KEY);
 
     queue.forEach((event, index) => {
+      // Always remove the event from storge.
```
Comment (weerd) [inline resources/assets/actions/event.js]: `storge` typo ;)

Comment (deadlybutter) [inline resources/assets/actions/event.js]: Its HTML6!

File: resources/assets/actions/event.js [hunk 2]
```diff
@@ -24,23 +24,26 @@ export function startQueue() {
...
+
       // Check if the event is over 30 min old before dispatching.
       const isValidTimestamp = isTimestampValid(event.createdAt, (30 * 60 * 1000));
 
       // Check if the user successfully authenticated
       const isAuthenticated = typeof getState().user.id !== 'undefined';
 
-      if (!isValidTimestamp || (event.requiresAuth && !isAuthenticated)) {
+      let shouldFireEvent = isValidTimestamp;
+      if (shouldFireEvent && event.requiresAuth) shouldFireEvent = isAuthenticated;
```
Comment (weerd) [inline resources/assets/actions/event.js]: can we stick w/ `{...}` brackets :)

File: resources/assets/actions/event.js (hunk 1 above)
Comment (weerd) [inline resources/assets/actions/event.js]: 🤔

**Pull request merged**